openstack-upi/openstack-upi.md

```diff
@@ -0,0 +1,281 @@
+want to use OpenStack UPI today have to fall back on
+the [bare metal][baremetal-upi] document or try to translate concepts
+from the AWS one.
+
+[baremetal-upi]: https://github.com/openshift/installer/blob/master/docs/user/metal/install_upi.md
+
+Similar to AWS and GCP, we want to provide the documentation and
+automation (via Ansible) that is specific to OpenStack -- using the
+terminology and tooling that OpenStack people are familiar with and
+highlighting considerations specific to that platform.
+
+## Motivation
+
+In addition to the general use cases supported by UPI (e.g. control
+over the creation of the resources of the cloud provider, custom
+Ignition configuration), there are additional reasons for having an
+OpenStack-specific UPI.
+
+First, unlike for completely hosted solutions such as AWS or GCP there
+is no single OpenStack.
+
+Some deployments are public clouds akin to AWS while others are only
+available within a single organisation, possibly disconnected from the
+internet. Moreover, the actual services and configuration can vary
+greatly between deployments.
+
+The OpenStack UPI should be able to support scenarios such as these:
+
+* Lack of Swift Object Storage (which we use for serving the bootstrap Ignition config as well as the registry storage)
+* Lack of floating IP addresses (which we use for the public API and ingress endpoints as well as for bootstrap log gathering via must-gather)
+* Disconnected from the internet
+* Desire to integrate all traffic with a load balancer or DNS that already exists within the organisation running the OpenStack
+
+### Goals
+
+* OpenStack UPI documentation available under:
+https://github.com/openshift/installer/blob/master/docs/user/openstack/install_upi.md
+* Ansible playbooks for automating the OpenStack resource creation available under:
+https://github.com/openshift/installer/tree/master/upi/openstack
+* CI job executing these playbooks running wherever the AWS and GCP jobs run
+* Optional documentation and playbook sections for Kuryr
+
+### Non-Goals
+
+Configurations and integrations that are not currently supported by the OpenStack IPI. For example:
+
+TODO
+
+Things such as detailed steps for the provider network support, using Octavia or Designate for the load balancer / DNS, custom storage integration and so on can be added as optional sections once the initial UPI work is complete.
+
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal actually is.
+
+* Write the Ansible playbook that automate the OpenStack resource creation (networks, subnets, ports, servers, etc.)
+* Add the Ansible dependencies to the OpenStack CI image
+* Write the UPI document, linking the Ansible templates
+* Create an UPI job in the [Release repository][openshift-release]
+* Add Kuryr documentation and playbooks as an optional SDN alternative
+
+[openshift-release]: https://github.com/openshift/release/
+
+### User Stories
```

Comment:
Should we add a user story for deploying with an alternate block storage or is it considered out of scope?

Reply:
For the first implementation of this I'd leave all that out of scope (other than maybe mentioning places where people can do this in the UPI doc).

Anything beyond the simplest deployment can be added later.